src/careamics/utils/metrics.py

```diff
@@ -32,84 +38,129 @@ def psnr(gt: np.ndarray, pred: np.ndarray, range: float = 255.0) -> float:
     ----------
-    x : NumPy array
+    x : Array
         Input array.
 
     Returns
     -------
-    NumPy array
+    Array
         Zero-mean array.
     """
-    return x - np.mean(x)
+    x = _torch_cast_to_double(x)
+    return x - x.mean()
 
 
-def _fix_range(gt: np.ndarray, x: np.ndarray) -> np.ndarray:
+def _fix_range(gt: Array, x: Array) -> Array:
     """
     Adjust the range of an array based on a reference ground-truth array.
 
     Parameters
     ----------
-    gt : np.ndarray
+    gt : Array
         Ground truth image.
-    x : np.ndarray
+    x : Array
         Input array.
 
     Returns
     -------
-    np.ndarray
+    Array
         Range-adjusted array.
     """
-    a = np.sum(gt * x) / (np.sum(x * x))
+    a = (gt * x).sum() / (x * x).sum()
     return x * a
 
 
-def _fix(gt: np.ndarray, x: np.ndarray) -> np.ndarray:
+def _fix(gt: Array, x: Array) -> Array:
     """
     Zero mean a groud truth array and adjust the range of the array.
 
     Parameters
     ----------
-    gt : np.ndarray
+    gt : Array
         Ground truth image.
-    x : np.ndarray
+    x : Array
         Input array.
 
     Returns
     -------
-    np.ndarray
+    Array
         Zero-mean and range-adjusted array.
     """
     gt_ = _zero_mean(gt)
     return _fix_range(gt_, _zero_mean(x))
 
 
-def scale_invariant_psnr(
-    gt: np.ndarray, pred: np.ndarray
-) -> Union[float, torch.tensor]:
+def scale_invariant_psnr(gt: Array, pred: Array) -> Union[float, torch.tensor]:
     """
     Scale invariant PSNR.
 
+    NOTE: `torch` does not support the `mean()` method for tensors whose
+    `dtype` is not `float`. Hence, this function will raise a warning and
+    automatically cast the input tensor to `float` if it is a `torch.Tensor`.
+
+    NOTE: results may vary slightly between `numpy` and `torch` due to the way
+    `var()` is computed. In `torch`, the unbiased estimator is used (i.e., SSE/n-1),
+    while in `numpy` the biased estimator is used (i.e., SSE/n).
+
     Parameters
     ----------
-    gt : np.ndarray
+    gt : Array
         Ground truth image.
-    pred : np.ndarray
+    pred : Array
         Predicted image.
 
     Returns
     -------
     Union[float, torch.tensor]
         Scale invariant PSNR value.
     """
-    range_parameter = (np.max(gt) - np.min(gt)) / np.std(gt)
-    gt_ = _zero_mean(gt) / np.std(gt)
+    # cast tensors to double dtype
+    gt = _torch_cast_to_double(gt)
+    pred = _torch_cast_to_double(pred)
+    # compute scale-invariant PSNR
+    range_parameter = (gt.max() - gt.min()) / gt.std()
+    gt_ = _zero_mean(gt) / gt.std()
     return psnr(_zero_mean(gt_), _fix(gt_, pred), range_parameter)
+
+
+def _torch_cast_to_double(x: Array) -> Array:
+    """
+    Cast a tensor to float.
+
+    Parameters
+    ----------
+    x : Array
+        Input tensor.
+
+    Returns
+    -------
+    Array
+        Float tensor.
+    """
+    if isinstance(x, torch.Tensor) and x.dtype != torch.float64:
+        warn(
```

Comment:
Is the warning necessary? I ask because it will never lead to any information loss. Also, we do not return this tensor; we just use it to compute the metrics.